src/pyramid_googleauth/routes.py

```diff
@@ -0,0 +1,30 @@
+"""The routes for the Pyramid app."""
+
+
+def add_routes(config):
+    """Register all routes."""
+
+    config.add_route(
+        "pyramid_googleauth_login",
+        config.registry.settings.get("pyramid_googleauth.login_route", "/ui/api/login"),
+    )
+    config.add_route(
+        "pyramid_googleauth_login_callback",
+        config.registry.settings.get(
+            "pyramid_googleauth.login_callback_route", "/ui/api/login_callback"
+        ),
+    )
+    config.add_route(
+        "pyramid_googleauth_logout",
+        config.registry.settings.get(
+            "pyramid_googleauth.logout_route", "/ui/api/logout"
+        ),
+    )
+
+    config.add_route(
+        "pyramid_googleauth_login_failure",
+        config.registry.settings.get(
+            "pyramid_googleauth.login_failure_route", "/ui/api/login_failure"
+        ),
+    )
```

Comment:
Can we put all the routes under `/googleauth/*` by default? That should avoid any potential conflict between the paths used by the extension by default and any paths that might be used by the app.

I'd like to avoid `/ui/*` as that's something I don't want to spread into our other apps. Generally speaking apps do usually put all their API endpoints under a `/api/` path (e.g. https://hypothes.is/api/) or a subdomain (e.g. https://api.github.com/). But apps don't usually put their entire UI under a `/ui/*` path. For example h UI pages like https://hypothes.is/search, https://hypothes.is/login, https://hypothes.is/users/seanh, https://hypothes.is/admin/, etc aren't all under `/ui/*`. Nor does GitHub put all its HTML pages under `/ui/*`. `/ui/*` is unusual and `/ui/api/*` is pretty weird, Checkmate shouldn't be doing it in my opinion.

Also renamed `/login_callback` and `/login_failure` to `/login/callback` and `/login/failure`.

```suggestion
    config.add_route(
        "pyramid_googleauth_login",
        config.registry.settings.get("pyramid_googleauth.login_route", "/googleauth/login"),
    )
    config.add_route(
        "pyramid_googleauth_login_callback",
        config.registry.settings.get(
            "pyramid_googleauth.login_callback_route", "/googleauth/login/callback"
        ),
    )
    config.add_route(
        "pyramid_googleauth_logout",
        config.registry.settings.get(
            "pyramid_googleauth.logout_route", "/googleauth/logout"
        ),
    )

    config.add_route(
        "pyramid_googleauth_login_failure",
        config.registry.settings.get(
            "pyramid_googleauth.login_failure_route", "/googleauth/login/failure"
        ),
    )
```